Add into_parts for `QSpi`

tgross35 opened the pull request:
# Summary
Add `into_parts` for `Qspi` to retreive pins

# Checklist
  - [ ] `CHANGELOG.md` for the BSP or HAL updated
  - [x] All new or modified code is well documented, especially public items
  - [x] No new warnings or clippy suggestions have been introduced (see CI or check locally)

## If Adding a new Board
  - [ ] Board CI added to `crates.json`
  - [ ] Board is properly following "Tier 2" conventions, unless otherwise decided to be "Tier 1"

## If Adding a new cargo `feature` to the HAL
  - [ ] Feature is added to the test matrix for applicable boards / PACs in `crates.json`

I was short on time and just needed a quick fix for my use case, but I figured I might as well open this PR to open discussion.

Is there any specific reason these methods to return the consumed pins/peripherals aren't available? Or is this a discussion that needs to happen at `embedded-hal`?. I've seen both `into_inner` and `into_parts` for naming, not sure which is more standard.

The quick fix I did here was for Qspi only, but the same goes for Spi, I2C, Uart, etc.

jbeaurivage commented:
@tgross35, I'm not sure every peripheral module adheres to this, but many of them call that type of function `free`.

jbeaurivage commented:
@tgross35 If you want to fix the formatting and rebase to master, I'll be happy to merge!

tgross35 commented:
Thanks @jbeaurivage, I totally forgot about this.

Changed the name to free and fixed the formatting issue. I don't know whether this is changelog-worthy or not, but let me know if so and I can add an entry.
